emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -0,0 +1,225 @@
+# emission/tests/funcTests/TestFunctionTiming.py
+
+import unittest
+import logging
+import time
+import typing as t
+import pymongo
+
+# Import the store_dashboard_time and store_dashboard_error functions
+import emission.storage.decorations.stats_queries as sdq
+
+# Import the existing Timer context manager
+import emission.core.timer as ec_timer
+
+# Import the database module for verification
+import emission.storage.timeseries.abstract_timeseries as esta
+import emission.core.get_database as edb
+
+class TestFunctionTiming(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        """
+        Set up resources before any tests are run.
+        """
+        logging.basicConfig(level=logging.INFO)
+        self.timeseries_db = esta.TimeSeries.get_time_series(None)
+
+    def tearDown(self):
+        """
+        Clean up relevant database entries after each test to maintain isolation.
+        """
+        # Define the metadata keys to clean
+        keys_to_clean = ["stats/dashboard_time", "stats/dashboard_error"]
+        try:
+            # Delete entries where metadata.key is in keys_to_clean from timeseries_db
+            edb.get_timeseries_db().delete_many({"metadata.key": {"$in": keys_to_clean}})
+            logging.info("Cleaned up database entries after test based on metadata keys.")
+        except Exception as e:
+            logging.error(f"Error cleaning up database entries: {e}")
+            raise
```

Comment:
If you are tearing down the database entries after every test, you will never end up with two entries, and will not be able to verify that the entries don't override each other.

You need to have a test that invokes multiple functions.

We are also not instrumenting only in chunks of one function at a time - we also want to instrument code blocks. Where is the test for that?